.github/workflows/typechecks.yml

```diff
@@ -0,0 +1,43 @@
+name: Tests, Linter & Typecheck
+
+on: pull_request
+
+jobs:  
+  lint:
+    name: 'Linting Markdown'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+
+      - name: Use Node.js 16.7
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16.7
+
+      - run: npm run prettier:md
+  test:
+    name: 'Typechecking'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Use Node.js 16.7
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16.7
+
+      - name: Install NPM dependencies
```

Comment:
The npm install and caching steps need to be added to the lint job above as well. Alternately move the lint section into the test section.